pkg/server/endpoint/endpoint.go

```diff
@@ -94,10 +98,14 @@ func (e *endpoint) ListenAndServe() error {
 		return err
```

Comment:
probably moot but this never releases the lock

Reply:
Oh... yes, thank you. I will fix this.